### PR TITLE
[MI-2971]: Fix checkbox and add size variants

### PR DESCRIPTION
**Pull request by SaurabhSharma-884**
### Ticket ID
https://brightscout.atlassian.net/browse/MI-2971

### Description
- Rebuild the checkbox component
- Add size variants for the checkbox

### Screenshot
Screenshot from 2023-04-06 17-20-00
Screenshot from 2023-04-06 17-20-11
Screenshot from 2023-04-06 17-20-22

### CODE REVIEW

File: src/components/CheckBox/CheckBox.component.tsx [hunk 1]
```diff
@@ -13,22 +16,44 @@ import {StyledCheckbox} from './CheckBox.styles';
  *  <Checkbox label="Label"/>
  * ```
  */
-export const Checkbox = (props:CheckBoxProps) => {
-    const {className = '', id, label, error, ...restProps} = props;
+export const Checkbox = (props: CheckBoxProps) => {
+    const {className = '', id, label, error, disabled, size = 'md', ...restProps} = props;
+
```
Comment (raghavaggarwal2308) [inline src/components/CheckBox/CheckBox.component.tsx:21]:
```suggestion
```

File: src/components/CheckBox/CheckBox.styles.ts [hunk 2]
```diff
@@ -38,55 +91,55 @@ export const StyledCheckbox = styled(Form.Check)({
...
-            color: colors.centerChannel,
-        },
+    '& .mm-checkbox-input[disabled] ~ .mm-checkbox-active': {
+        backgroundColor: colors.centerChannel_48,
+    },
 
-        '&:checked': {
-            backgroundColor: colors.error,
-        },
+    // Styles of input checkbox active on checked state
```
Comment (raghavaggarwal2308) [inline src/components/CheckBox/CheckBox.styles.ts:141]:
```suggestion
    // Styles of input checkbox active in checked state
```

File: src/components/CheckBox/CheckBox.component.tsx [hunk 3]
```diff
@@ -13,22 +16,44 @@ import {StyledCheckbox} from './CheckBox.styles';
  *  <Checkbox label="Label"/>
  * ```
  */
-export const Checkbox = (props:CheckBoxProps) => {
-    const {className = '', id, label, error, ...restProps} = props;
+export const Checkbox = (props: CheckBoxProps) => {
+    const {className = '', id, label, error, disabled, size = 'md', ...restProps} = props;
```
Comment (raghavaggarwal2308) [inline src/components/CheckBox/CheckBox.component.tsx:20]: constant for 'md', 'sm' & 'lg'?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

File: src/constants/index.ts [hunk 4]
```diff
@@ -1,5 +1,14 @@
+import {CheckboxSizeTypes} from '@Components/CheckBox/CheckBox';
+
+const CHECKBOX_SIZE: Record<string, CheckboxSizeTypes> = {
```
Comment (manojmalik20) [inline src/constants/index.ts:3]: It should be an enum.

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**